### PR TITLE
Play nice(r) by lazy-init'ing the flush loop

### PR DESCRIPTION
**Pull request by bensigelman**
This fixes an underlying issue discovered in a Tornado environment.

### CODE REVIEW

**Comment by bensigelman** [comment]
My only concern here is that we're not going far enough to protect against `fork()` calls... that said, defending against them *in general* would basically require `getpid()` calls to be sprinkled anywhere one might want to normally acquire a mutex, and down that path lies madness. (Let's just hope we're not on that path??)

File: lightstep/recorder.py [hunk 1]
```diff
@@ -99,13 +99,26 @@ def __init__(self,
...
                 'Runtime(periodic_flush_seconds={0}) means we will never flush to lightstep unless explicitly requested.'.format(
                     self._periodic_flush_seconds))
-        else:
+
+    def _maybe_init_flush_thread(self):
+        """Start a periodic flush mechanism for this recorder if:
+
+        1. periodic_flush_seconds > 0, and 
+        2. self._flush_thread is None *or* the pid has changed (indicating a
+           fork() of some sort)
```
Comment (ross) [inline lightstep/recorder.py]: Comment here mentions a pid change, but there's no explicit checkin for it below. That something you were looking to explicitly handle?

Comment (jmacd) [inline lightstep/recorder.py]: This is a fair question: I suppose the concern is use of fork() not followed by exec()?

Comment (bensigelman) [inline lightstep/recorder.py]: @ross this was an oversight. I was previously doing the `os.getpid()` thing here, but the more I thought about it the more I realized how insufficient the local check was against arbitrary `fork()` calls. Hence the scaled-back version that I have here.

@jmacd yes, exactly. And this is an idiom used by `tornado` at startup (as well as other python server frameworks like `gunicorn`).

Anyway, I've updated the comment to match the code. After that I'll merge and push out a new artifact.

**Pull request merged**